app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -96,29 +101,48 @@ class CheckAgencyController @Inject()(
         .bindFromRequest()
         .fold(
           formWithErrors => {
+            if (formWithErrors.errors.exists(_.message == "error.business-type-value.invalid")) {
+              Logger.warn("Select business-type form submitted with invalid identifier")
+              throw new BadRequestException("submitted form value did not contain valid businessType identifier")
+            }
             Future successful Ok(html.check_business_type(formWithErrors))
           },
-          businessType => Future successful Redirect(routes.CheckAgencyController.checkAgencyStatus())
+          validatedBusinessType =>
+            Future successful Redirect(
+              routes.CheckAgencyController.showCheckAgencyStatus(validatedBusinessType.businessType))
         )
     }
   }
 
-  def showCheckAgencyStatus: Action[AnyContent] = Action.async { implicit request =>
+  def showCheckAgencyStatus(businessType: Option[String]): Action[AnyContent] = Action.async { implicit request =>
     withSubscribingAgent { implicit agent =>
-      withMaybeContinueUrlCached {
-        mark("Count-Subscription-CheckAgency-Start")
-        Future successful Ok(html.check_agency_status(CheckAgencyController.knownFactsForm))
+      businessType match {
+        case Some(businessTypeIdentifier)
+            if CheckAgencyController.validBusinessTypes.contains(businessTypeIdentifier) => {
+          withMaybeContinueUrlCached {
```

Comment:
This `withMaybeContinueUrl` should be moved to the outside of the match on `businessType`. The reason being, if people are coming from say the trusts registration flow then they'll still be pointing to this endpoint with a `continue` parameter but without a `businessType` parameter. We should still take their 'continueUrl' parameter and cache it before redirecting them to the `/check-business-type` page.